### DISCUSSION TITLE
Ansible roles and collections

### DISCUSSION THREAD

**Discussion by stoeps13** [HCL-TECH-SOFTWARE/connections-automation#37] [General]
Hi,

the repository does a good job getting a running Connections, but it needs some adjustments or even larger rewrites to get it working in real (grown over some years) environments.

May I suggest to split the repository and make [collections and roles ](https://www.ansible.com/blog/hands-on-with-ansible-collections) on galaxy.ansible.com, so everybody can grab parts and reuse it their playbooks?

In the moment cloning the whole repository and changing adding the stuff for the environment is doable, but it is nearly unpossible to get updates integrated in such custom playbooks. With the usage of requirements.yml and just loading the roles and collections which are needed (sometimes you need no databases, because they're already provided). On the other hand it is easier to add additional services, like special Kubernetes releases, DNS and so on through the Galaxy community roles/collections.

More informations and examples:

- [https://docs.ansible.com/ansible/latest/user_guide/collections_using.html](https://docs.ansible.com/ansible/latest/user_guide/collections_using.html)
- [https://docs.ansible.com/ansible/latest/dev_guide/migrating_roles.html](https://docs.ansible.com/ansible/latest/dev_guide/migrating_roles.html)
- [https://docs.ansible.com/ansible/latest/dev_guide/developing_collections.html](https://docs.ansible.com/ansible/latest/dev_guide/developing_collections.html)
- [https://www.jeffgeerling.com/blog/2020/ansible-best-practices-using-project-local-collections-and-roles](https://www.jeffgeerling.com/blog/2020/ansible-best-practices-using-project-local-collections-and-roles)
- [https://github.com/geerlingguy/ansible-collection-k8s](https://github.com/geerlingguy/ansible-collection-k8s)

Regards
Christoph

**Comment by dejo1307** [comment]
Hey Christoph,

This definitely makes sense.

To put some light on this - current repo is, obviously, monorepo, so each new commit ensures that everything works end to end. The road towards separating it more then it is right now (hcl vs third_party folders) was paved by simply separating this. 

I've took this with the team. Since it is already open source, and everyone can change it and customise, it is hard to prioritise it high against all the new stuff being added and happening. Our main goal with this is to ensure that it covers our features, and it works for all our versions we are covering. 

We are still assessing how and if to do this refactoring because, as much as I understand what RedHat is trying to do with introducing those concepts, we need to weight if this is going to explode complexity on our side as well (testing end to end each commit vs different components version matrix). But some refactoring will happen for sure, and I will update as soon as we have general plan and next steps (which would be sooner then probably expected).

Btw. thanks for a great idea, really appreciated.

Regards,
Dejan
